docs/guide/api-javascript.md

```diff
@@ -30,6 +30,10 @@ const { createServer } = require('vite')
 })()
 ```
 
+::: tip NOTE
+When using `createServer` and `build` in the same Node.js process, both functions rely on `process.env.NODE_ENV` to work properly, which also depends on the `mode` config option. To prevent conflicting behaviour, set `process.env.NODE_ENV` or the `mode` of the two APIs to `development`. Otherwise, you can spawn a child process to run the APIs separately.
```

Comment:
```suggestion
When using `createServer` and `build` in the same Node.js process, both functions rely on `process.env.\u200bNODE_ENV` to work properly, which also depends on the `mode` config option. To prevent conflicting behavior, set `process.env.\u200bNODE_ENV` or the `mode` of the two APIs to `development`. Otherwise, you can spawn a child process to run the APIs separately.
```

Reply:
@bluwy not ideal, but at least until there is a better solution we can do this.

Reply:
We can leave the issue as is also though, since the prod build works correctly. It'll just be a bit annoying when developing locally.

I don't mind using this hack if we're not releasing a new docs version soon though. And we can find a fix at the meantime.

Also, this issue also happens in the SSR page, which uses `process.env.NODE_ENV` too

Reply:
We can use the hack there too then, we aren't going to release the docs for another month. And Kia just told me he is going to send a PR with a new theme so I prefer that he don't deal with these errors in dev.